skbase/tests/test_meta.py

```diff
@@ -168,3 +168,23 @@ def test_metaestimator_composite(long_steps):
 
     meta_est.set_params(bar__b="something else")
     assert meta_est.get_params()["bar__b"] == "something else"
+
+
+def test_basemetaobject_set_params_uses_baseobject_reset():
+    """Test that BaseMetaObject.set_params correctly triggers BaseObject.reset."""
+
+    class ResetCheckMetaObject(BaseMetaObject):
+        def __init__(self, a=1, steps=None):
+            self.a = a
+            self.steps = steps if steps is not None else []
```

Comment:
this is not compliant with the API - self params should never be changed.

There should also be `steps_` etc, and we should also do `get_params` in the test.